geode-core/src/test/java/org/apache/geode/cache/management/MemoryThresholdsDUnitTest.java

```diff
@@ -522,17 +521,29 @@ public Object call() throws Exception {
     verifyListenerValue(server1, MemoryState.EVICTION, 2, true);
     verifyListenerValue(server1, MemoryState.NORMAL, 1, true);
 
-    LogWriterUtils.getLogWriter().info("before NORMAL->CRITICAL->NORMAL");
-    // NORMAL -> EVICTION -> NORMAL
-    server2.invoke(new SerializableCallable("NORMAL->CRITICAL->NORMAL") {
+    // NORMAL -> CRITICAL
+    server2.invoke(new SerializableCallable("NORMAL->CRITICAL") {
       public Object call() throws Exception {
         GemFireCacheImpl gfCache = (GemFireCacheImpl) getCache();
         gfCache.getInternalResourceManager().getHeapMonitor().updateStateAndSendEvent(950);
+        return null;
+      }
+    });
+
+    verifyListenerValue(server2, MemoryState.CRITICAL, 2, true);
+    verifyListenerValue(server2, MemoryState.EVICTION, 3, true);
+    verifyListenerValue(server2, MemoryState.NORMAL, 1, true);
+    verifyListenerValue(server1, MemoryState.CRITICAL, 2, true);
+    verifyListenerValue(server1, MemoryState.EVICTION, 3, true);
+    verifyListenerValue(server1, MemoryState.NORMAL, 1, true);
+
+    server2.invoke(new SerializableCallable("CRITICAL->NORMAL") {
+      public Object call() throws Exception {
+        GemFireCacheImpl gfCache = (GemFireCacheImpl) getCache();
```

Comment:
Recommend changing from GemFireCacheImpl to InternalCache. The getInternalResourceManager method should exist on InternalCache.